lib/gooddata/core/loggers/context_logger_decorator.rb

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2010-2018 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+module GoodData
+  # Logger decorator with ability to enrich the message with brick context
+  class ContextLoggerDecorator < SimpleDelegator
+    # entry-point
+    # @param [Logger] logger logger to decorate
+    def initialize(logger)
+      @logger = logger
+    end
+
+    # log methods to be decorated
+    def add(severity, message = nil, progname = nil)
+      @logger.add(severity, enrich(message, GoodData.context_manager.context), progname)
+    end
+
+    private
+
+    # Enrich given message.
+    # @param message  [String] or [Hash] message to enrich
+    # @return masked_message [String] or [Hash] enriched message
+    def enrich(message, context)
```

Comment:
info: Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction) (reek)